client/extensions/woocommerce/app/products/product-create.js

```diff
@@ -93,22 +94,29 @@ class ProductCreate extends React.Component {
 	onSave = () => {
 		const { site, product, translate } = this.props;
 
-		const successAction = () => {
+		const successAction = ( products ) => {
+			const newProduct = head( products );
 			page.redirect( getLink( '/store/products/:site', site ) );
-
 			return successNotice(
 				translate( '%(product)s successfully created.', {
 					args: { product: product.name },
 				} ),
-				{ duration: 4000, isPersistent: true }
+				{
+					displayOnNextPage: true,
+					duration: 8000,
+					button: translate( 'View' ),
+					onClick: () => {
+						window.open( newProduct.permalink );
```

Comment:
It's too bad global notices don't seem to support `external`, that'd make this a bit simpler (and more accessible) because it would just be a plain link tag 🙁  (no action for you here, just an observation - `onClick`s always set off my accessibility-radar)